cpp/src/arrow/util/basic_decimal.cc

```diff
@@ -451,6 +463,27 @@ static int64_t FillInArray(const BasicDecimal128& value, uint32_t* array,
   return 1;
 }
 
+/// Expands the given value into an array of ints so that we can work on
```

Comment:
Comment that the output array is in big-endian.

Reply:
Done. Thank you!